### ISSUE TITLE
prettier does not work on windows

### ISSUE THREAD

**Issue by luoluoqixi** [oedotme/generouted#105]
### Describe the Bug

prettier does not work on windows

plugin source code:
```
plugins/react-router/src/plugin/generate.ts
...
const prettier = './node_modules/.bin/prettier'
...
```

running './node_modules/.bin/prettier' exec on windows fails
```'.' is not recognized as an internal or external command```

because '.\\node_modules\\.bin\\prettier' should be used on windows

reference:
https://stackoverflow.com/questions/20765337/how-to-fix-is-not-an-internal-or-external-command-error

temporary fix:

```scripts/plugins/generouted```
```typescript
import fs from 'fs';
import path from 'path';

export default async () => {
  let revert = () => {};
  if (process.platform === 'win32') {
    const replacePrettier = (filePath: string): (() => void) => {
      const search = /\.\/node_modules\/\.bin\/prettier/g;
      const replace = '.\\\\node_modules\\\\.bin\\\\prettier';
      const oldStr = fs.readFileSync(filePath, { encoding: 'utf8' });
      const newStr = oldStr.replace(search, replace);
      fs.writeFileSync(filePath, newStr);
      revert = () => {
        fs.writeFileSync(filePath, oldStr);
      };
      return revert;
    };
    const fileExtends = ['.js', '.cjs'];
    const generoutedPath = path.join(process.cwd(), './node_modules/@generouted/react-router/dist/plugin');
    const files = fs.readdirSync(generoutedPath);
    const revers = [];
    for (let i = 0; i < files.length; i++) {
      const file = files[i];
      if (fileExtends.some((f) => file.toLocaleLowerCase().endsWith(f))) {
        revers.push(replacePrettier(path.join(generoutedPath, file)));
      }
    }
    revert = () => {
      for (let i = 0; i < revers.length; i++) {
        revers[i]();
      }
    };
  }
  const generouted = (await import('@generouted/react-router/plugin')).default;
  revert();
  return generouted;
};
```
```vite.config.ts```
```typescript
import generoute from './scripts/plugins/generouted';

export default defineConfig(async () => {
  const generouted = await generoute();
  return {
    plugins: [
      react(),
      generouted({
        output: 'src/routes.tsx',
        format: true,
      }),
  };
});
```

### Generouted Version

1.15.1

### Your Example Website or App or Reproduction

nothing

### Steps to Reproduce the Bug or Issue

1. pnpm add prettier
2. run `pnpm run dev` on windows using vite
3. src/router.ts is generated, but prettier is not executed

### Expected Behavior

prettier is execution

### Platform

- OS: Windows10
- Browser: Chrome
- Browser version: 110.0

**Comment by oedotme** [comment]
Hey @luoluoqixi, thanks for reporting this.

I've pushed a fix for this issue, at [v1.15.2](https://github.com/oedotme/generouted/releases/tag/v1.15.2). It was a pretty small fix, by wrapping the prettier bin path with `path.resolve`:

```diff
+ import path from 'path'

-  const prettier = './node_modules/.bin/prettier'
+  const prettier = path.resolve('./node_modules/.bin/prettier')
```

I tested it on a windows machine and it worked just fine, here's the commit reference 9644c96

Please let me if that resolved the issue for you.

**Comment by luoluoqixi** [comment]
upgraded to 1.15.2, the prettier works normally, thanks~

> 嘿@luoluoqixi，感谢您报告此事。
> 
> [我已在v1.15.2](https://github.com/oedotme/generouted/releases/tag/v1.15.2)中修复了此问题。这是一个相当小的修复，通过用以下内容包装更漂亮的 bin 路径`path.resolve`：
> 
> ```diff
> + import path from 'path'
> 
> -  const prettier = './node_modules/.bin/prettier'
> +  const prettier = path.resolve('./node_modules/.bin/prettier')
> ```
> 
> 我在 Windows 机器上测试了它，它工作得很好，这是提交参考[9644c96](https://github.com/oedotme/generouted/commit/9644c96931e955db05a4481ee72a27387851a67d)
> 
> 如果这为您解决了问题，请告诉我。
